### PR TITLE
Update to latest feature-intent and feature-recentlyclosed APIs (and update Android Components).

### PR DESCRIPTION
**Pull request by pocmo**
Fixing the breaking changes that AC introduced.

### CODE REVIEW

**Comment by pocmo** [comment]
One UI test failed with:

> android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: recently_closed_tabs.uuid (code 1555 SQLITE_CONSTRAINT_PRIMARYKEY)

I'll investigate and fix.

**Comment by pocmo** [comment]
The correct fix for this is to also do what is described in the following issue:
https://github.com/mozilla-mobile/android-components/issues/8497

I'll implement in AC and update this PR with the new Nightly version once available.

File: app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedController.kt [hunk 1]
```diff
@@ -64,23 +65,23 @@ class DefaultRecentlyClosedController(
...
-                activity.openToBrowser(
-                    from = BrowserDirection.FromRecentlyClosed
-                )
-            }
+    override fun handleRestore(item: RecoverableTab) {
+        tabsUseCases.restore(item)
+
+        store.dispatch(
+            RecentlyClosedAction.RemoveClosedTabAction(item)
+        )
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedController.kt:81]: Not sure if later we could add a use case for this

**Review by Amejia481 — APPROVED** [review]
Looks great ✨ , just one comment.

File: app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedController.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/library/recentlyclosed/RecentlyClosedController.kt:81]: Yeah, probably, I guess ideally all action dispatching is hidden behind a nicer use case. I guess we could have a specific restore use case that calls into the restore above and then dispatches this additionally.

**Comment by bb010g** [comment]
> One UI test failed with:
> 
> > android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: recently_closed_tabs.uuid (code 1555 SQLITE_CONSTRAINT_PRIMARYKEY)
> 
> I'll investigate and fix.

This exception has been crashing Fenix when tabs from previous launches are (attempted) to be closed.

```
E ExceptionHandler: Uncaught exception handled:
E ExceptionHandler: android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: recently_closed_tabs.uuid (code 1555 SQLITE_CONSTRAINT_PRIMARYKEY)
```

**Pull request merged**